docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,184 @@
+
+    def deepchecks_collate(data) -> BatchOutputFormat:
+        # Extracting images and label and converting images of (N, C, H, W) into (N, H, W, C)
+        images = torch.stack([x[0] for x in data]).permute(0, 2, 3, 1)
+        labels = [x[1] for x in data]
+        # Convert ImageNet format images into to [0, 255] range format images.
+        mean, std  = [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
+        images = np.clip(std * images.numpy() + mean, 0, 1) * 255
+        return BatchOutputFormat(images= images, labels= labels)
+
+    data_loader = DataLoader(my_dataset, batch_size=64, generator=torch.Generator(),
+                             collate_fn=deepchecks_collate)
+    vision_data = VisionData(data_loader, task_type='classification')
+    vision_data.head()
+
+From TensorFlow Dataset
+-----------------------
+There are two possible ways to create a deepchecks compatible tensorflow
+`Dataset object <https://www.tensorflow.org/api_docs/python/tf/data/Dataset>`_. You can either create it
+in a way that directly outputs the data in the required format or convert an existing dataset,
+we will demonstrate the latter.
+
+Assume a tensorflow dataset object that outputs a batch of images and labels as a tuple of (images, labels).
+We will use the `map <https://www.tensorflow.org/api_docs/python/tf/data/Dataset#map>`_
+function to convert the data into :doc:`Deepchecks' format </user-guide/vision/supported_tasks_and_formats>`.
+
+.. code-block:: python
+
+    from deepchecks.vision import VisionData, BatchOutputFormat
+
+    def deepchecks_map(batch) -> BatchOutputFormat:
+        # Extracting images and label and converting images of (N, C, H, W) into (N, H, W, C)
+        images = batch[0].permute(0, 2, 3, 1)
+        labels = batch[1]
+        # Convert ImageNet format images into to [0, 255] range format images.
+        mean, std  = [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
+        images = np.clip(std * images.numpy() + mean, 0, 1) * 255
+        return BatchOutputFormat(images= images, labels= labels)
+
+    deepchecks_dataset = my_dataset.map(deepchecks_map)
+    vision_data = VisionData(deepchecks_dataset, task_type='classification')
+    vision_data.head()
+
+
+
+Adding Model Predictions
+========================
+Some of deepchecks tests, including the :doc:`model evaluation checks and suite </checks_gallery/vision>`,
+require model predictions in
+order to run. Models prediction are supplied via the batch loader in a similar fashion to the images and labels.
+There are several ways to supply them which can be divide into two categories: Pre-calculated predictions and
+on-demand inference.
+
+On-demand Inference
+-------------------
+In this case we will need to incorporate the model object in the relevant format transformation function. This can be
+done either by using the model as a global variable, creating a wrapper class for the transformation function or
```

Comment:
```suggestion
done either by using the model as a global variable, creating a wrapper class for the collate/map function or
```

Reply:
I think that "transformation function" is a scary phrase so suggested just to name the common ones.